core/constraints/validation.go

```diff
@@ -215,20 +219,90 @@ func (v *validator) checkValidValues(cons Value) error {
 	return nil
 }
 
+// ValidatorInvalidConstraintErr represents an error that occurs
+// when a validation constraint is violated.
+// It provides details for the
+// invalid inputs, suggested valid values and all possible valid values.
+type ValidatorInvalidConstraintErr struct {
```

Comment:
I'd just call this `InvalidVocabValueErr` or something like that.
The error is specific to an invalid vocab value being used and the name `ValidatorInvalidConstraintErr` is too generic.

Reply:
I see, updated and thanks Ian!